File: server/ExpenseTracker/Models/Category.cs
namespace ExpenseTracker.Models
{
    public class Category : BaseEntity
    {
        public string Name { get; set; }
        public string Description { get; set; }
        public Guid UserId { get; set; }
        public User User { get; set; }
        public ICollection<Expense> Expenses { get; set; }
        public ICollection<RecurringExpense> RecurringExpenses { get; set; }
    }
}

File: client/src/Interfaces/Category.ts
import { User } from './User';
import { Expense } from './Expense';
import { RecurringExpense } from './RecurringExpense';

export interface Category {
  id: string; // Assuming BaseEntity has an ID
//   createdAt?: Date; // Assuming BaseEntity has timestamps
//   updatedAt?: Date; // Assuming BaseEntity has timestamps
  name: string;
  description: string;
  userId: string;
  user: User;
  expenses: Expense[];
  recurringExpenses: RecurringExpense[];
}
